src/mapper.ts

```diff
@@ -263,80 +33,44 @@ export class Mapper {
     let kstat: KeyValue[] = [];
     let currentStat: KeyValue[] = [];
 
-    if(seriesList[0].columns !== undefined) {
-      let keys = seriesList[0].columns.map(col => col.text);
-
-      const keyColumn = this._panelConfig.getValue('keyColumn');
-      let keyIndex = 0;
-      if(keyColumn !== '') {
-        keyIndex = keys.findIndex(key => key === keyColumn);
-      }
 
-      const title = keys[keyIndex];
-
-      let skipIndexes: number[] = [keyIndex];
-      const skipColumn = this._panelConfig.getValue('skipColumn');
-      if(skipColumn !== '') {
-        skipIndexes.push(keys.findIndex(key => key === skipColumn));
-      }
+    let keys = seriesList[0].columns.map(col => col.text);
 
-      const maxValue = _.max(
-        seriesList[0].rows.map(
-          row => _.sum(
-            row.filter((value, idx) => !_.includes(skipIndexes, idx))
-          )
-        )
-      );
-
-      const filteredKeys = keys.filter((key, idx) => !_.includes(skipIndexes, idx));
-      
-      // TODO: it's wrong, we return a bad type here
-      return seriesList[0].rows.map(
-        row => new MultiProgressItem(
-          this._panelConfig,
-          row[keyIndex],
-          filteredKeys,
-          row.filter((value, idx) => !_.includes(skipIndexes, idx)),
-          maxValue as number
-        )
-      );
+    const keyColumn = this._panelConfig.getValue('keyColumn');
+    let keyIndex = 0;
+    if(keyColumn !== '') {
+      keyIndex = keys.findIndex(key => key === keyColumn);
     }
 
-    if(mappingType === 'datapoint to datapoint') {
-      if(statType === StatType.TOTAL && seriesList.length == 1) {
-        kstat = this._mapKeysTotal(seriesList);
-      } else {
-        kstat = this._mapNumeric(seriesList, statType, nullMapping);
-      }
-      currentStat = this._mapNumeric(seriesList, StatType.CURRENT, nullMapping);
-    } else {
-      kstat = this._mapTargetToDatapoints(seriesList, statType);
-      currentStat = this._mapTargetToDatapoints(seriesList, StatType.CURRENT);
+    const title = keys[keyIndex];
```

Comment:
`keyIndex` can be `-1`

Reply:
good news that we don't use this `title` variable so I am going to just remove the line